sympy/parsing/latex/lark/latex_parser.py

```diff
@@ -109,6 +130,13 @@ def parse_latex_lark(s: str):
     """
     if _lark is None:
         raise ImportError("Lark is probably not installed")
+
+    global _personal_macros
+    global _lark_latex_parser
+    if macros and (_personal_macros != macros):
+        _personal_macros = macros
+        _lark_latex_parser = LarkLaTeXParser(macros=macros)
+
```

Comment:
Can you elaborate a bit on this part? I don't understand why `global _personal_macros` and `global _lark_latex_parser` is needed here.

Reply:
Honestly, I always forget if it is necessary to use the `global` keyword in order to get/set global variables, and at that moment I just put them there in order to proceed with more important aspects of the code, and it worked. Later, I forgot to test if I actually need them. Do you think I should remove them?

---

Explanation of the code: note that `.doparse()` does not allow you to add new terminals to the Grammar or modify existing terminals in the Grammar; therefore you need to create a new instance of `LarkLaTeXParser` every time `parse_latex_lark()` is called with a new set of custom macros, which is why the current values of the macros are being stored in `_personal_macros` and a new instance of `LarkLaTeXParser` is being created only when `_personal_macros` changes.

Reply:
I see. That makes sense.

It's my bad for not being comfortable with the `global` keyword in Python.